app/src/components/map/components/MapFeatureList.tsx

```diff
@@ -37,6 +37,17 @@ const MapFeatureList: React.FC<MapFeatureListProps> = (props) => {
     }
   };
 
+  /**
+   * When editing an existing project or plan there may be some existing features
+   * that need to be updated. This effect will recalculate the feature ids.
+   */
+  useEffect(() => {
```

Comment:
Instead of running a useEffect which will re-trigger on every update of this component.
Why not just recalculate the FeatureIds when you are adding them to the list?

Reply:
I am... But I've found with some edge cases existing spatial data, on component initialization, the IDs are out of order.

Reply:
I think this can be handled more elegantly still, having useEffects with any trigger ie: `,[]` can be dangerous for infinite rerenders. Especially when you are resetting your own Prop

Reply:
Looking at Leaflets source code... Think I can bring that logic internal.... and remove leaflet entirely.

Reply:
What you are experiencing may be a side effect of the initial map rerendering a few times. In reference to your bug ticket. I would avoid patching this problem with more renders and have data calculated properly on insert.

Reply:
Hmmm... Wondering if we can drop this though. In theory it's only the seed data that's doing this.... Might need some testing.